packages/vscode-extension/lib/preview.js

```diff
@@ -38,10 +51,18 @@ export function preview(component) {
   }
 
   const key = `preview:/${component._source.fileName}:${component._source.lineNumber}`;
+
+  const lastPreview = global.__RNIDE_previews.get(key);
+
   global.__RNIDE_previews.set(key, {
```

Comment:
not really, the component after refresh is a "new" component, therefore the state will be dropped. Typically, this is handled by hot reload runtime which makes it possible for different component to retain state, but in this case hot reload doesn't see it because the component instance isn't created within a regular render scope (much like hot reload can't properly handle updates to the root view)